src/detect-engine-address-ipv4.c

```diff
@@ -845,12 +844,13 @@ static int DetectAddressIPv4TestAddressCmp01(void)
 
     DetectAddressFree(a);
     DetectAddressFree(b);
-    return result;
+    FAIL_IF_NOT(result);
+    PASS;
 
  error:
     DetectAddressFree(a);
     DetectAddressFree(b);
-    return 0;
+    FAIL;
```

Comment:
Basically, all the lines  that do something and then use a `goto error` are a `FAIL` case, and should be converted to a `FAIL_IF` format.

One of the goals for the FAIL/PASS API conversion is to remove all branching logic (the `if` checks) in unit tests, too.

So let's try to clean up this test (and others in this file if they follow a similar patter), so that we first have all `FAIL` possibilities, and, if nothing has failed, we will then have the calls to Free functions, to avoid memory leaks on `PASS`, and finish the test with a `PASS`.

This will further improve readability, and also allow us to get rid of the `result` variable, since we will be explicitly failing for each check case.

Reply:
Ok, thank you so much for clarifying! I was feared it could cause unwanted memory leaks by not freeing the variables in case the test fails, I don't need to worry with that?

Reply:
Indeed, we only have to avoid memory leaks in the `PASS` scenario :)